File: db4oj/core/src/com/db4o/foundation/MutableInt.java
/* Copyright (C) 2006  db4objects Inc.  http://www.db4o.com */

package com.db4o.foundation;


/**
 * @exclude
 */
public class MutableInt {
    
    private int _value;
    
    public MutableInt() {
        
    }
    
    public MutableInt(int value) {
        _value = value;
    }
    
    public void add(int addVal){
        _value += addVal;
    }
    
    public void substract(int substractVal){
        _value -= substractVal;
    }
    
    public int value(){
        return _value;
    }

}

File: db4on/core/src/core/com/db4o/foundation/MutableInt.cs
namespace com.db4o.foundation
{
	/// <exclude></exclude>
	public class MutableInt
	{
		private int _value;

		public MutableInt()
		{
		}

		public MutableInt(int value)
		{
			_value = value;
		}

		public virtual void Add(int addVal)
		{
			_value += addVal;
		}

		public virtual void Substract(int substractVal)
		{
			_value -= substractVal;
		}

		public virtual int Value()
		{
			return _value;
		}
	}
}
